dts/bindings/stepper/zephyr,gpio-stepper.yaml

```diff
@@ -3,33 +3,29 @@
 # SPDX-License-Identifier: Apache-2.0
 
 description: |
-  GPIO Stepper Controller cluster for darlington transistor arrays or dual H-bridge
+  GPIO Stepper Controller for darlington transistor arrays or dual H-bridge
 
   Example:
     /* Lead A is connected Lead C and Lead B is connected to Lead D*/
-    stepper {
-        compatible = "zephyr,gpio-steppers";
-        motor: motor {
-            gpios = <&gpioa 9 GPIO_ACTIVE_HIGH>,  /* Lead A1/A */
-                    <&gpioc 7 GPIO_ACTIVE_HIGH>,  /* Lead B1/B */
-                    <&gpiob 0 GPIO_ACTIVE_HIGH>,  /* Lead A2/C */
-                    <&gpioa 7 GPIO_ACTIVE_HIGH>;  /* Lead B2/D */
-        };
+    stepper: stepper {
+        compatible = "zephyr,gpio-stepper";
+        gpios = <&gpioa 9 GPIO_ACTIVE_HIGH>,  /* Lead A1/A */
+                <&gpioc 7 GPIO_ACTIVE_HIGH>,  /* Lead B1/B */
+                <&gpiob 0 GPIO_ACTIVE_HIGH>,  /* Lead A2/C */
+                <&gpioa 7 GPIO_ACTIVE_HIGH>;  /* Lead B2/D */
     };
 
-compatible: "zephyr,gpio-steppers"
+compatible: "zephyr,gpio-stepper"
 
-child-binding:
-  description: GPIO Controller for stepper motor
-  include:
-    - name: stepper-controller.yaml
-      property-allowlist:
-        - micro-step-res
-        - invert-direction
+include:
+  - name: stepper-controller.yaml
+    property-allowlist:
+      - micro-step-res
+      - invert-direction
```

Comment:
Don't know if this is common practice for drivers generally or if it's fine that if more fields are added they simply go unused?

Reply:
you are right atleast in this case all the properties are used. hence have removed `property_allowlist`